src/edu/csus/ecs/pc2/core/standings/ScoreboardUtilites.java

```diff
@@ -188,6 +190,95 @@ public static ScoreboardJsonModel createContestStandingsFromJSON(File jsonFile)
     public static int toInt(String string, int defaultNumber) {
         return Utilities.nullSafeToInt(string, defaultNumber);
     }
-    
-    
+
+    /**
+     * 
+     * @param clientId
+     * @param contest
+     * @return
+     */
+    public static Run[] getRunsForUserDivision(ClientId clientId, IInternalContest contest) {
+
+        String division = getDivision(contest, clientId);
+        
+//        System.out.println("debug 22 getRunsForUserDivision for "+clientId+" div is "+division);
+
+        if (ClientType.Type.TEAM.equals(clientId.getClientType())) {
+
+            List<Run> teamRuns = new ArrayList<Run>();
+            for (Run run : contest.getRuns()) {
+
+                ClientId runClientId = run.getSubmitter();
+
+                if (runClientId.equals(clientId)) {
+                    // add team/client's own runs
+                    teamRuns.add(run);
+                } else {
+                    // add if submitting team in same division
+                    if (matchDivsion(contest, division, run.getSubmitter())) {
+                        teamRuns.add(run);
+//                        System.out.println("debug 22 Added run " + run);
+                    }
+                }
+            }
+
+            return (Run[]) teamRuns.toArray(new Run[teamRuns.size()]);
+            
+        } else {
+            return contest.getRuns();
+        }
+    }
+
+    protected static boolean matchDivsion(IInternalContest contest, String inputDivision, ClientId submitter) {
+        
+          String division = getDivision(contest, submitter);
+          
+//          System.out.println("debug 22 matchDivsion  "+inputDivision+" vs "+division+" for "+submitter);
+          
+          if (inputDivision == null && division == null) {
+              return true;
+          } if (inputDivision == null) {
+              return false;
+          } else {
+              return inputDivision.equals(division);
+          }
+    }
+
+    /**
+     * Return division for input clientId.
+     * 
+     * @param contest
+     * @param submitter
+     * @return null if no division, else a digit
+     */
+    public static String getDivision(IInternalContest contest, ClientId submitter) {
+        
+        ElementId groupId = contest.getAccount(submitter).getGroupId();
+        if (groupId == null) {
+            return null;
+        }
+        
+        Group group = contest.getGroup(groupId);
+        if (group == null) {
+            return null;
+        }
+        
+        String groupName = group.getDisplayName().trim();
+        
+        return getDivision(groupName);
+    }
+
+    public static String getDivision(String groupName) {
+
+        int idx = groupName.lastIndexOf('D');
```

Comment:
It is very scary that we're putting something as "hard-coded and deeply hidden" as "Group names must end with Dx" into the core code.  At _least_ it seems like this should be flagged as something intended to be temporary, hopefully with a reference to an Issue number for fixing it.

Reply:
it's not clear to mean why the code from the DSA that dealt with finding the division number was not re-used?
```
+        // TODO this bit should in the future can probably go away when divisions are supported better.
+        int lastFound = 0;
+        for (Group group : groups) {
+            // no reference to groups that should not be displayed on scoreboard
+            if (!group.isDisplayOnScoreboard()) {
+                continue;
+            }
+            Pattern pattern = Pattern.compile("D\\d+$");
+            Matcher matcher = pattern.matcher(group.getDisplayName());
+            if (matcher.find()) {
+                Integer found = Integer.parseInt(matcher.group(0).substring(1));
+                if (found.intValue() > lastFound) {
+                    lastFound = found.intValue();
+                }
+                divisionIndexHash.put(group, found);
+            }
+        }
+        divisionCount = lastFound;
```

Reply:
not aware of that code, until now.   No plans to use that code because the entire division/group is gong to be redesigned/written.   I will add a comment that the above is temporary and reference the issue added to redesign how divisions are done.    Added https://github.com/pc2ccs/pc2v9/issues/689 for this purpose.

Reply:
> 
The whole D# is kludgy and should be temporary and will be done someday.